deploifai/project/create.py

```diff
@@ -2,17 +2,129 @@
 
 from deploifai.context import pass_deploifai_context_obj, DeploifaiContextObj
 from deploifai.utilities import local_config
+from deploifai.api import DeploifaiAPI, DeploifaiAPIError
+from deploifai.utilities.user import parse_user_profiles
+from PyInquirer import prompt
 
 
 @click.command()
-@pass_deploifai_context_obj
 @click.argument("name")
-def create(context: DeploifaiContextObj, name: str):
+@click.option("--workspace", help="Workspace name", type=str)
+@pass_deploifai_context_obj
+def create(context: DeploifaiContextObj, name:str, workspace):
     """
     Create a new project
     """
-    project_id = 'project_id'
+    if not context.is_authenticated():
+        click.echo("Login using deploifai login first")
+        raise click.Abort()
```

Comment:
Refactor this part into a reusable function because this can be used in many different commands that require authorization.

Reply:
where to put this reusable code?

Reply:
Using an `is_authenticated` decorator to check if the user is authenticated to run a particular command